godot-macros/src/docs/markdown_converter.rs

```diff
@@ -5,91 +5,337 @@
-    };
+                // Prefix each line with "> "
+                let mut out = String::new();
+                for (i, line) in content.split("[br]").enumerate() {
+                    if i > 0 {
+                        out.push_str("[br]");
+                    }
+                    out.push_str("> ");
+                    out.push_str(line);
+                }
+                Some(out)
+            }
 
-    Some(bbcode)
-}
+            // Code block -> [codeblock lang=??]...[/codeblock]
+            Code(md::Code { value, lang, .. }) => {
+                let maybe_lang = lang
+                    .as_ref()
+                    .map(|l| format!(" lang={}", l))
+                    .unwrap_or_default();
+                Some(format!("[codeblock{maybe_lang}]{value}[/codeblock]"))
+            }
 
-/// Calls [`walk_node`] over every node it receives, joining them with the supplied separator.
-fn walk_nodes(nodes: &[md::Node], definitions: &HashMap<&str, &str>, separator: &str) -> String {
-    nodes
-        .iter()
-        .filter_map(|n| walk_node(n, definitions))
-        .collect::<Vec<_>>()
-        .join(separator)
+            // List -> each item is on its own line with indentation.
+            // For ordered lists, we use a counter we increment for each item.
+            // For unordered lists, we use '•'.
+            List(lst) => {
+                let indent = " ".repeat(level * 4);
+                let is_ordered = lst.ordered;
+                let mut counter = lst.start.unwrap_or(1);
+
+                let mut lines = Vec::new();
+                for item_node in &lst.children {
+                    if let md::Node::ListItem(item) = item_node {
+                        // Converts the item's children. These may be paragraphs or sub-lists, etc.
+                        // We join multiple paragraphs in the same item with [br].
+                        let item_str = self.walk_nodes_as_block(&item.children, level + 1);
+                        let bullet = if is_ordered {
+                            let b = format!("{}.", counter);
+                            counter += 1;
+                            b
+                        } else {
+                            "•".to_string()
+                        };
+                        let checkbox = if let Some(checked) = item.checked {
+                            if checked {
+                                "[x] "
+                            } else {
+                                "[ ] "
+                            }
+                        } else {
+                            ""
+                        };
+
+                        lines.push(format!("{indent}{bullet} {checkbox}{item_str}"));
+                    }
+                }
+                // Each item is separated by [br]
+                if lines.is_empty() {
+                    None
+                } else {
+                    Some(lines.join("[br]"))
+                }
```

Comment:
this is (at least, i didnt keep close track) the second time this pattern has come up, maybe we can extract it to its own function? that'd let us do like
`join_br(lines)?` especially if we early return on `None`

Reply:
Did similar with `join_if_nonempty`